luisy/targets.py

```diff
@@ -137,6 +163,172 @@ def _try_to_upload(self, overwrite=False):
         )
 
 
+class CloudTarget(LuisyTarget):
+
+    def __init__(self, path, **kwargs):
+        LuisyTarget.__init__(
+            self,
+            path=path,
+            format=None,
+            is_tmp=False
+        )
+        self.kwargs = kwargs
+
+    def _try_to_upload(self, overwrite=False):
+        """No need to upload cloud targets."""
+        pass
+
+    def _try_to_download(self):
+        """No need to download cloud targets."""
+        pass
+
+
+class SparkTarget(CloudTarget):
+    """
+    this abstract class is for targets working with spark instances.
+    """
+    @property
+    def spark(self):
+        try:
+            return Config().spark
+        except AttributeError:
+            raise AttributeError(
+                "spark session was not found in config. Make sure to have all the databricks "
+                "parameters set in order to start the spark session!"
+            )
+
+
+class DeltaTableTarget(SparkTarget):
+    # TODO: Can we get rid of fileending
+    file_ending = 'DeltaTable'
+
+    def __init__(
+            self,
+            outdir=None,
+            schema="schema",
+            catalog="catalog",
+            table_name=None
+    ):
+        self.outdir = outdir
+        self.table_name = table_name
+        self.schema = schema
+        self.catalog = catalog
+
+    def make_dir(self, path):
+        # TODO: Nothing to do here, adapt interface?
+        pass
+
+    def remove(self):
+        self.spark.sql(f"DROP TABLE IF EXISTS {self.table_uri}")
+
+    @property
+    def table_uri(self):
+        return f"{self.catalog}.{self.schema}.{self.table_name}"
+
+    @property
+    def path(self):
+        # TODO: Path here is more an identifier that shows up in `.luisy.hashes`
+        return os.path.join(
+            self.outdir,
+            f"{self.table_uri}.{self.file_ending}",
+        )
+
+    def exists(self):
+        """
+        Checks whether the Deltatable exists.
+
+        Note:
+            Ideally, we would call `self.spark.catalog.tableExists(self.table_uri)` to check whether
+            the table exists, but this always returns `False`.
+        """
+        try:
+            self.spark.sql(f"SELECT 1 from {self.table_uri}")
+            return True
+        except AnalysisException:
```

Comment:
when I run a Task with a DeltaTable on a databricks notebook, and I create the Table for the first time, I get an error here,
since the Exception is slightly different from the one you imported here.
I get an "pyspark.errors.exceptions.captured.AnalysisException",
but you import an pyspark.errors.exceptions.connect.AnalysisException

- I am using databricks runtime 13.3

Reply:
Thanks for reporting! Looks like both exception exist and but both are essentially re-imports of `pyspark.errors.exceptions.base.AnalysisException`. Will fix that!

Reply:
perfect, thanks for fixing this!